### PR TITLE
CI improvements

### PR DESCRIPTION
**Pull request by hamidelmaazouz**
- Adds semi-automatic version control. Basically allows one to tag latest main
- Adds a conditional to publish only when a tag with a specific format is pushed to main

### CODE REVIEW

File: .github/workflows/package-publish.yml
Comment (jfriel-oqc) [inline .github/workflows/package-publish.yml:6]: want + on the major version, too! Let's be optimistic and assume qat will be around for more than 9 major versions!

File: .github/workflows/package-publish.yml
Comment (jfriel-oqc) [inline .github/workflows/package-publish.yml:125]: this can be removed, too. I can now set up the project to directly connect to github and not use tokens. Just needed the project to be created in the first instance

File: .github/workflows/cla-assistant.yaml
Comment (jfriel-oqc) [inline .github/workflows/cla-assistant.yaml:7]: why remove this name?

File: .github/workflows/version.yaml
Comment (jfriel-oqc) [inline .github/workflows/version.yaml:1]: very nice :+1:

**Review by jfriel-oqc — APPROVED** [review]
some minor points, looks great, thanks Hamid. Approving to unblock as the comments are minor

**Review by jfriel-oqc — APPROVED** [review]
some minor points, looks great, thanks Hamid. Approving to unblock as the comments are minor

File: .github/workflows/cla-assistant.yaml
Comment (hamidelmaazouz) [reply .github/workflows/cla-assistant.yaml:7]: it's only moved to the top of file ^^ pedantic me on a Friday :)

File: .github/workflows/package-publish.yml
Comment (hamidelmaazouz) [reply .github/workflows/package-publish.yml:125]: will leave this to another improvement sweep, we can actually create a reusable action similar to other projects

File: .github/workflows/cla-assistant.yaml
Comment (jfriel-oqc) [reply .github/workflows/cla-assistant.yaml:7]: somehow missed that in the diff... makes sense!

**Pull request merged**
